prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -470,15 +471,16 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             .register(self.mir.span, ErrorCtxt::Unexpected);
         let method_with_fold_unfold = foldunfold::add_fold_unfold(
             self.encoder,
-            self.cfg_method,
+            self.cfg_method.clone(),
```

Comment:
Is the clone needed just to read the name of the method later? If so, you can store the method's name in a local variable before this call instead of cloning.